### PR TITLE
Janica 1st try

### PR DESCRIPTION
**Pull request by JanicaTang**
creating....

### CODE REVIEW

File: factorial.py [hunk 1]
```diff
@@ -6,13 +6,30 @@
...
+def factorial(n):
+    result = 1
+    for i in range(1, n+1):
+        result = result * i
+
+    return result
+
 
 def test_factorial():
     assert_equal(factorial_recursive(1), 1)
```
Comment (yarikoptic) [inline factorial.py]: it is always good to test boundary conditions (i.e. 0 value) and "incorrect" values (e.g. -1 ;) ) -- what will happen? ;)

File: factorial.py [hunk 2]
```diff
@@ -6,13 +6,30 @@
...
+    else:
+        result = n * factorial_recursive(n-1)
+    return result
+
+
+# no recursion version of n!
+def factorial(n):
+    result = 1
+    for i in range(1, n+1):
+        result = result * i
```
Comment (yarikoptic) [inline factorial.py]: for such cases, ```result *= i``` is preferable

File: factorial.py [hunk 3]
```diff
@@ -6,13 +6,30 @@
 
 from nose.tools import assert_equal
 
+
 def factorial_recursive(n):
```
Comment (yarikoptic) [inline factorial.py]: I was blurbing about docstrings in the class for a reason ;)

File: factorial.py (hunk 1 above)
Comment (JanicaTang) [inline factorial.py]: oh, interesting thought...
Just tried -1 and got an error saying "runtime maximum recursion exceeded" and also with a thousand lines of running messages :(

Comment (yarikoptic) [inline factorial.py]: >  >  def test_factorial():
>  >      assert_equal(factorial_recursive(1), 1)

>    oh, interesting thought...
>    Just tried -1 and got an error saying "runtime maximum recursion exceeded"
>    and also with a thousand lines of running messages :(

yeap ;)

simplest solution -- state your expectations in the function with e.g.
"assess" statement and a good message to go along ;)

then you could catch those "exceptions" and test for them using 

from nose.tools import assert_raises

-- just check documentation for assert_raises on how to use it
